subsys/net/ip/dhcpv4.c

```diff
@@ -1243,6 +1243,17 @@ void net_dhcpv4_stop(struct net_if *iface)
 	k_mutex_unlock(&lock);
 }
 
+void net_dhcpv4_restart(struct net_if *iface)
+{
+	net_dhcpv4_stop(iface);
```

Comment:
The `net_dhcpv4_stop()` function will issue a `NET_EVENT_IPV4_DHCP_STOP` event, but the restart will not issue a `NET_EVENT_IPV4_DHCP_START` afterwards. It also seems that we're missing some additional actions here (randomizing `xid`, registering net_mgmt callback).

Perhaps it'd be better to reuse the existing codebase, i.e. move `net_dhcpv4_start()` functionality into a static function, with additional `timeout` parameter, and then from `net_dhcpv4_start()` call it with a random timeout, while from `net_dhcpv4_restart()` just setting the timeout to 0?

Reply:
Sure, makes sense.